### ISSUE TITLE
Do the benchmarks have a final held-out test set?

### ISSUE THREAD

**Issue by efirdc** [brain-score/brain-score#303]
Hello,

I read through the brain-score paper today and I had a question about the train/testing methodology. My understanding is that for every feature representation in a model, an n-fold cross-validation is performed, and then the features with the highest mean score across the n-folds are selected. My question is whether the score across the n-folds for the selected features is reported, or if the corresponding n models are run on a final held-out test set. In the former case, I would be concerned that the selected features are a bit over-fit to the cross-validation folds. Especially in the very deep models that have many feature representations, there would be a lot of opportunities to find over-fitting features.

Thank you,
Cory

**Comment by mschrimpf** [comment]
Hi Cory,

Thanks for your message! Since the 2018 bioRxiv paper, we have actually updated the methodology to address exactly the concern that you outlined: We now evaluate models that need to search over layers in two stages with separate data (these are also described abstractly in the [2020 Neuron paper](https://www.cell.com/neuron/fulltext/S0896-6273(20)30605-X)):
1. **Convert to BrainModel**: If a model does not commit to the particulars of the ventral stream such as cortical regions V1 through IT, we call it a *BaseModel* and run a [standard procedure](https://github.com/brain-score/model-tools) to make those commitments. Specifically, we test the similarity of every layer of the BaseModel with public splits of data from all regions -- for instance, we search which layer best matches public data from IT. This layer is then *committed* which means it's locked in and cannot be changed anymore. When all these commitments are done, we call it a *BrainModel* (see also the [corresponding interface](https://github.com/brain-score/brain-score/blob/master/brainscore/model_interface.py)). BrainModels can be interacted with ~like a primate subject, e.g. record from V1 or IT, or perform a particular task.
2. **Evaluation of BrainModel on held-out data**: The [brain-score](https://github.com/brain-score/brain-score) benchmarks themselves interact only with BrainModels such that there is no more freedom in e.g. the choice of layers. This also means that the same layer will be used for all IT predictions, rather than one layer for IT benchmark 1 and another layer for IT benchmark 2 (as often happens when BaseModels are tested in different papers). The data in the evaluation benchmarks (leading to the scores shown on the website) is separate from the data used for the BrainModel conversion. 

Finally, models could still benefit from repeated evaluation on the evaluation benchmarks (e.g. submit variations of a model multiple times), but as we are adding more and more benchmarks, any such overfitting should be exposed.

Let me know if that makes sense!
Martin

**Comment by efirdc** [comment]
Hi,

Thanks very much for the detailed explanation, I understand how it works now. 

It's interesting that the very deep models are still mostly dominating the leaderboard. I suppose the best features found through cross-validation are actually generalizing to the test set.

Thanks again,
Cory

**Comment by mschrimpf** [comment]
Yes indeed -- from the leaderboard though, it does seem like some deep architectures are better than others. For instance, almost all of the current top 20 models have skip connections. Other deep models such as nasnet, inception, or also the recent vision transformers underperform in comparison.

The CORnet-S model is more shallow yet retains high performance on Brain-Score (as well as ImageNet), you might be interested in that.
